test/semantics/select-case.f90

```diff
@@ -0,0 +1,176 @@
+   CHARACTER(KIND=2), PARAMETER    :: se = 'c'
+   CHARACTER(KIND=4), PARAMETER    :: mh='d'
+
+   select case (grade1)
+      case ('A')
+      print*, "Excellent!"
+      case ('B')
+      case ('C')
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade2)
+      case (1)
+      print*, "Excellent!"
+      case (2)
+      case (3)
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case (.true.)
+       name = 'true'
+      case (.false.)
+        name = 'false'
+   end select
+   print*, "Your grade is ", name
+
+   select case (name)
+      case default
+         print*, "Invalid grade"
+      case ('now')
+      case ('test')
+         print*, "Well done"
+   end select
+
+   !ERROR: SELECT CASE expression must be of type CHARACTER, INTEGER, OR LOGICAL
+   select case (grade4)
+      case (1.0)
+      print*, "Excellent!"
+      case (2.0)
+      case (3.0)
+         print*, "Well done"
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case default
+         print*, "Invalid grade"
+      case (.true.)
+         print*, "Well done"
+      !ERROR: Not more than one of the selectors of SELECT CASE statement may be DEFAULT
+      case default
+         print*, "Invalid grade"
+   end select
+
+   select case (grade3)
+      case default
+         print*, "Invalid grade"
+      case (.true.)
+      !ERROR: CASE value must be of type LOGICAL
+      case (3)
+         print*, "Well done"
+   end select
+
+   select case (grade3)
+      case default
+         print*, "Invalid grade"
+      !ERROR: SELECT CASE expression of type LOGICAL must not have range of case value
+      case (.true. :)
+   end select
+
+   select case (grade2)
+      case default
+         print*, "Invalid grade"
+      case (2 :)
+         print*, "Well done"
+      !ERROR: CASE value must be of type INTEGER
+      case (.true. :)
+   end select
+
+   select case (grade3)
+      case (.true.)
+       name = 'true'
+      case (.false.)
+        name = 'false'
+      !ERROR: CASE value true matches a previous CASE statement
+      case (.true.)
+       name = 'true'
+      !ERROR: CASE value false matches a previous CASE statement
+      case (grade5)
+       name = 'true'
+   end select
+   print*, "Your grade is ", name
+
+   select case (grade2)
+      case (100:)
+         print*, "Excellent!"
+      case (:30)
+         print*, "Excellent!"
+      case (40)
+         print*, "Excellent!"
+      case (90)
+         print*, "Excellent!"
+      case (91:99)
+         print*, "Very good!"
+      !ERROR: CASE value 81:90 matches a previous CASE statement
+      case (81:90)
+         print*, "Very good!"
+      !ERROR: CASE value :80 matches a previous CASE statement
+      !ERROR: CASE value :80 matches a previous CASE statement
```

Comment:
Here is how the error looks as of now.

select-case.f90:123:7: error: CASE value 81:90 matches a previous CASE statement
        case (81:90)
        ^^^^^^^^^^^^
select-case.f90:118:7: Previous CASE statement matching 81:90
        case (90)
        ^^^^^^^^^
Error is consecutive and has the same case value in both the error message to avoid any confusion.
Since we already have the same case value in both the error message, adding the line number to the error message might look like too much of duplicate info in error messages.
Let me know your opinion.

Reply:
Agree, ignore my comment.